Lib/pathlib.py

```diff
@@ -920,18 +874,18 @@ def samefile(self, other_path):
             other_st = self.__class__(other_path).stat()
         return os.path.samestat(st, other_st)
 
+    @abstractmethod
     def iterdir(self):
         """Iterate over the files in this directory.  Does not yield any
         result for the special paths '.' and '..'.
         """
-        for name in os.listdir(self):
-            yield self._make_child_relpath(name)
+        raise NotImplementedError
 
     def _scandir(self):
         # bpo-24132: a future version of pathlib will support subclassing of
         # pathlib.Path to customize how the filesystem is accessed. This
         # includes scandir(), which is used to implement glob().
```

Comment:
Should this read: `_scandir`?
Also is the comment obsolete now, or is the version proposed by this PR not yet the mentioned future version?